manifests/init.pp

```diff
@@ -282,138 +282,142 @@
-  Optional[Integer[0]] $dhcp_mclt = $foreman_proxy::params::dhcp_mclt,
-  Optional[Integer[0, 255]] $dhcp_load_split = $foreman_proxy::params::dhcp_load_split,
-  Optional[Integer[0]] $dhcp_load_balance = $foreman_proxy::params::dhcp_load_balance,
+  Optional[Stdlib::Port] $dhcp_failover_port = 519,
+  Optional[Integer[0]] $dhcp_max_response_delay = 30,
+  Optional[Integer[0]] $dhcp_max_unacked_updates = 10,
+  Optional[Integer[0]] $dhcp_mclt = 300,
+  Optional[Integer[0, 255]] $dhcp_load_split = 255,
+  Optional[Integer[0]] $dhcp_load_balance = 3,
   Boolean $dhcp_manage_acls = $foreman_proxy::params::dhcp_manage_acls,
-  Boolean $dns = $foreman_proxy::params::dns,
-  Foreman_proxy::ListenOn $dns_listen_on = $foreman_proxy::params::dns_listen_on,
-  Boolean $dns_managed = $foreman_proxy::params::dns_managed,
-  String $dns_provider = $foreman_proxy::params::dns_provider,
+  Boolean $dns = false,
+  Foreman_proxy::ListenOn $dns_listen_on = 'https',
+  Boolean $dns_managed = true,
+  String $dns_provider = 'nsupdate',
   String $dns_interface = $foreman_proxy::params::dns_interface,
   Optional[Stdlib::Fqdn] $dns_zone = $foreman_proxy::params::dns_zone,
-  Optional[Variant[String, Array[String]]] $dns_reverse = $foreman_proxy::params::dns_reverse,
-  String $dns_server = $foreman_proxy::params::dns_server,
-  Integer[0] $dns_ttl = $foreman_proxy::params::dns_ttl,
+  Optional[Variant[String, Array[String]]] $dns_reverse = undef,
+  String $dns_server = '127.0.0.1',
+  Integer[0] $dns_ttl = 86400,
   String $dns_tsig_keytab = $foreman_proxy::params::dns_tsig_keytab,
   String $dns_tsig_principal = $foreman_proxy::params::dns_tsig_principal,
-  Array[String] $dns_forwarders = $foreman_proxy::params::dns_forwarders,
-  String $libvirt_network = $foreman_proxy::params::libvirt_network,
-  String $libvirt_connection = $foreman_proxy::params::libvirt_connection,
-  Boolean $bmc = $foreman_proxy::params::bmc,
-  Foreman_proxy::ListenOn $bmc_listen_on = $foreman_proxy::params::bmc_listen_on,
-  Enum['ipmitool', 'freeipmi', 'shell', 'ssh'] $bmc_default_provider = $foreman_proxy::params::bmc_default_provider,
-  String $bmc_ssh_user = $foreman_proxy::params::bmc_ssh_user,
-  Stdlib::Absolutepath $bmc_ssh_key = $foreman_proxy::params::bmc_ssh_key,
-  String $bmc_ssh_powerstatus = $foreman_proxy::params::bmc_ssh_powerstatus,
-  String $bmc_ssh_powercycle = $foreman_proxy::params::bmc_ssh_powercycle,
-  String $bmc_ssh_poweroff = $foreman_proxy::params::bmc_ssh_poweroff,
-  String $bmc_ssh_poweron = $foreman_proxy::params::bmc_ssh_poweron,
-  Boolean $realm = $foreman_proxy::params::realm,
-  Foreman_proxy::ListenOn $realm_listen_on = $foreman_proxy::params::realm_listen_on,
-  String $realm_provider = $foreman_proxy::params::realm_provider,
+  Array[String] $dns_forwarders = [],
+  String $libvirt_network = 'qemu:///system',
+  String $libvirt_connection = 'default',
```

Comment:
I think these 2 parameters have got switched.

Should be
```puppet
String $libvirt_network = 'default'
String $libvirt_connection = 'qemu:///system'
``??

Reply:
Indeed. They were swapped in order in params.pp. Updated now.